docs/branching_model_team_agreement.md

```diff
@@ -0,0 +1,18 @@
+# Branching Model Team Agreement
+
+This document lists the agreed upon practices the team will be using. As they are discussed and changed, update this file accordingly.
+
+## Guardrails
+- The default branch for this repo is `main`
+- all commits must be made to a non-protected branch and submitted via a pull request before they can be merged into `main`. (There can be no direct commits made to `main`.)
+- Pull requests require at least 1 approval before the can be merged to `main`. Any commits made to an approved branch will dismiss the approval. (If you commit new code to a branch after it gets approved but before you merge it, you have to get it approved again.)
+- Pull requests must pass CI checks in order to be merged to `main`
+- Branches are automatically deleted when merged
+- Squash merge has been disabled
+
+## Practices
+- Make draft pull requests early and often to facilitate a transparent process
+- Reviews can be done by the first person who gets to it. If the code needs explaining, request the author to walk you through it
+- Follow peer review best practices by suggesting opportunities to improve code during peer review, merging as soon as the code is better than the code in the target branch and release ready
+- Treat any opportunity for improvement feedback identified during peer review but not implemented in the PR where it was raised as technical debt worthy of a new issue referencing the PR where the comments first came up
+- Branch naming policy is as follows: dm-issue#-brief-description
```

Comment:
Is the `dm-` prefix performing some meaningful function? (per the example of this branch being `dm-14-branching-model-agreement`) Or could we do with a more succinct `issue#-brief-description`?

Reply:
In the upstream there's a pattern (maybe team agreement) of beginning each branch with the JIRA ticket number. Perhaps we continued that practice in this repo? A team agreement on consistent naming would be helpful...though there's often a one-to-many relation between an issue and the PRs that close it. Perhaps we could consider including the ticket number we're closing in the description section of the commit that substantially delivers the feature? For single-commit pull requests (or draft PRs that begin after pushing the very first commit) the title and description of the commit message are auto populated into the github PR title and description.

Reply:
The `dm-` prefix is something we're doing in the upstream, so we decided to keep it the same here

Reply:
```suggestion
- Branch naming policy is as follows: brief-description
```

Reply:
Perhaps there's a risk that by keeping it the same as upstream we will confuse references to JIRA ticket numbers (the DM-xx form) with the Github Issues of this fork? In standup @BCerki asked if the suggestion above was a typo. The option of beginning branches with the issue number they relate to `issue#-brief-description` is another possible team agreement, and the impact of that decision would be to create a mechanism whereby we avoid making pull requests that don't correspond to tracked issues. As we were discussing in standup, sometimes a quick refactor is best enabled by directly sending a pull request without creating an issue–sometimes trivial refactors unblock our feature tickets. Additionally, I'd note that our branches are a transitory artifact–per suggestion above migrating the issue linkage to commit messages themselves would make this ephemeral linkage permanent and immutable.

Reply:
Marked this unresolved until we fully migrate discussion to #21 so it shows when people follow link from that issue.